NPE when opening certain notifications

```
2021-01-18 12:58:25.965 E/AndroidRuntime: FATAL EXCEPTION: main
    Process: crux.bphc.cms, PID: 15642
    java.lang.RuntimeException: Unable to start activity ComponentInfo{crux.bphc.cms/crux.bphc.cms.activities.CourseDetailActivity}: java.lang.NullPointerException
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3270)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.NullPointerException
        at crux.bphc.cms.activities.CourseDetailActivity.resolveIntent(CourseDetailActivity.kt:62)
        at crux.bphc.cms.activities.CourseDetailActivity.onCreate(CourseDetailActivity.kt:39)
        at android.app.Activity.performCreate(Activity.java:7824)
        at android.app.Activity.performCreate(Activity.java:7813)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1306)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 
```

Certain notifications may not have a course id or the course may not be available locally. In such cases, the app should gracefully fetch the new app content in the MainActivity or CourseDetailActivity. 
Additionally, MainActivity should handle the `Home page` course which contains the Site News announcements.

--- a/app/src/main/java/crux/bphc/cms/activities/CourseDetailActivity.kt
+++ b/app/src/main/java/crux/bphc/cms/activities/CourseDetailActivity.kt
@@ -21,7 +21,7 @@ import io.realm.Realm
 class CourseDetailActivity : AppCompatActivity() {
     private lateinit var realm: Realm
 
-    private var course: Course? = null
+    private lateinit var course: Course
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
@@ -52,17 +52,17 @@ class CourseDetailActivity : AppCompatActivity() {
             courseId = enrolCourse!!.id
         }
 
-        course = realm
-                .where(Course::class.java)
-                .equalTo("id", courseId)
-                .findFirst()
+        this.course = realm
+            .where(Course::class.java)
+            .equalTo("id", courseId)
+            .findFirst() ?: Course(courseId)
 
         // check if enrolled
-        if (course == null) {
-            setCourseEnrol(enrolCourse!!)
-            title = enrolCourse!!.shortName
+        if (courseId == -1 && enrolCourse != null) {
+            setCourseEnrol(enrolCourse)
+            title = enrolCourse.shortName
         } else {
-            title = course!!.shortName
+            title = course.shortName
 
             val url = Uri.parse(contextUrl) ?: Uri.EMPTY
             if (Urls.isCourseSectionUrl(url) || Urls.isCourseModuleUrl(url)) {
@@ -96,7 +96,7 @@ class CourseDetailActivity : AppCompatActivity() {
         val fragmentTransaction = supportFragmentManager.beginTransaction()
         val courseSectionFragment = CourseContentFragment.newInstance(
                 UserAccount.token,
-                course!!.id,
+                course.id,
                 contextUrl,
         )
         fragmentTransaction.replace(
@@ -111,7 +111,7 @@ class CourseDetailActivity : AppCompatActivity() {
         setCourseContentFragment("")
         supportFragmentManager.executePendingTransactions()
         val fragmentTransaction = supportFragmentManager.beginTransaction()
-        val forumFragment: Fragment = ForumFragment.newInstance(course!!.id, forumId, course!!.shortName)
+        val forumFragment: Fragment = ForumFragment.newInstance(course.id, forumId, course.shortName)
         fragmentTransaction.addToBackStack(null)
                 .replace(R.id.course_section_enrol_container, forumFragment, "Announcements")
         fragmentTransaction.commit()
@@ -125,10 +125,10 @@ class CourseDetailActivity : AppCompatActivity() {
         supportFragmentManager.executePendingTransactions()
         val fragmentTransaction = supportFragmentManager.beginTransaction()
         val discussionFragment: Fragment = DiscussionFragment.newInstance(
-            course!!.id,
+            course.id,
             forumId,
             discussionId,
-            course!!.shortName
+            course.shortName
         )
         fragmentTransaction.addToBackStack(null)
                 .replace(R.id.course_section_enrol_container, discussionFragment, "Discussion")

